vllm/v1/core/scheduler.py

```diff
@@ -544,6 +562,24 @@ def update_from_output(
             new_logprobs = None
             new_token_ids: List[int] = []
 
+            # Handle guided decoding FSM advancement if applicable
+            if (request.use_guided_decoding and request.grammar
+                    and request.is_grammar_ready):
```

Comment:
If we're handling model output, is_grammar_ready should always be true and request.grammar should always exist, so those conditions can be dropped, I think (just keep use_guided_decoding)